src/Internal/Payment/State/StateFactory.php

```diff
@@ -57,7 +57,9 @@ public function create_state( string $state_class, PaymentContext $context ): Ab
 		}
 
 		$state = $this->container->get( $state_class );
+		$context->update_state_transition( str_replace( __NAMESPACE__ . '\\', '', $state_class ) );
```

Comment:
Is it necessary to remove the namespace from the class name? It would make sense to remove it for the logs, but IMO we should save the full class name within the context object.

The reason: We are logging those changes as a string for now, but we might need to store this in the database, and reconstruct it for for other log formats.

Reply:
Done, including full name for now